### ISSUE TITLE
plugin crashes/dosent work in the latest arch release 0.27.2

### ISSUE THREAD

**Issue by structnull** [ItsDrike/hyprland-dwindle-autogroup#5]
- toggle group isn't grouping as it should group.
- attached this video on how it isn't working
https://litter.catbox.moe/1sfj6h.mp4
- crashes when ungrouping
    - coredumpctl --> https://paste.rs/duCaY
    -  hyprland crash log --> https://paste.rs/ijaVU, https://paste.rs/mzDpU

hyprctl version -->
```
Hyprland, built from branch  at commit b08b72358ad549fd066e5be0fc3aa4c9df367607 dirty ().
Tag: v0.27.2

flags: (if any)
```

ps : This time I recompiled the plugin against the corresponding hyprland release.

**Comment by ItsDrike** [comment]
Hey, thanks for reporting this, I have also encountered an issue on v0.27.2, though for some reason it was slightly different to what you reported, and I haven't experienced any hard crashes, for me I only noticed that when I grouped 2 windows, moving the cursor to where the child window (that should now be groupped under the parent) was somehow shown that window through the group.

 I haven't yet had the time to investigate this further, but I will give it a look later today, or some time tomorrow.

**Comment by structnull** [comment]
Hey thanks for the prompt response!. 
now I changed to the latest hyprland-git version it seems to be working again albeit with graphical gliches like this.

- ![image](https://github.com/ItsDrike/hyprland-dwindle-autogroup/assets/87846149/2e56806c-8fda-4140-99d4-0c332817b0af)
- ![image](https://github.com/ItsDrike/hyprland-dwindle-autogroup/assets/87846149/178eb104-0fa4-4225-b395-04dd0ece191e)
-  ![image](https://github.com/ItsDrike/hyprland-dwindle-autogroup/assets/87846149/89ba8d75-c62f-4872-b89f-2f5785f55b2b)

Also there hasn't anymore hard crashes after switching to the git branch but it works weirdly and sometimes clients refuse to be in a grouped tab.

```
hyprctl version :
Hyprland, built from branch main at commit 375e8385eeabbcdae197e392394849e2cbb6f3ed dirty (Fix background occlusion ignoring monitor position (2771)).
Tag: v0.27.2-32-g375e8385

flags: (if any)
```
Thanks again for looking into this

**Comment by ItsDrike** [comment]
Yeah, that glitch is the issue I was talkig about. When you hover over the right side, the window actually shows up event though it's supposed to be in a group.

I already did a bisect and found the hyprland commit that caused this, but I'm not home rn so I'll look into writing up a fix in later when I get back.

Btw, the reason the plugin worked without recompilation is essentially a mirracle. The Hyprland ABI is not stable, and can't be relied on, if nothing changes in the files we're using, then it generally works, but any change to compilation arguments, or compiling with different tool, or indeed a change in the original source could cause really weird problems which are almost impossible to debug, so recompiling is crucial every time, don't just trust that nothing will go wrong.

You could probably set up a pacman hook if you're forgetting, that could auto recompile after hyprland gets updated.

**Comment by ItsDrike** [comment]
Should be fixed in https://github.com/ItsDrike/hyprland-dwindle-autogroup/commit/c1c883a237c58b8736f1f46f383490f0621b136c

Edit: I've also did a pretty nice cleanup, improving the performance in 120a9c3. This was only possible as with this hyprland change, they finally allow moving windows into a group without also focusing them. I'm not sure how noticable the performance improvement actually is, but it's certainly cleaner behavior from the plugin.

In case you'd still have issues, do try to also build from the original fix (without the cleanup commit), just in case the issue would be with the cleanup code, but from my testing, everything has worked perfectly.

**Comment by structnull** [comment]
Hey Thanks for the quick fix! . Now its working great with the latest git , now being snappier than before. 

Although there is a niche nitpick that the group title bar disappear when a client is moved inside (the moved client's titlebar only) the tab. This behavior isn't observed when moving an already tabbed client into another tabbed client.Maybe this is a hyprland issue or on how the title-bars displayed using the default behavior.

attached recording for reference.
-  https://github.com/ItsDrike/hyprland-dwindle-autogroup/assets/87846149/fce2aa5b-5edb-491a-825b-760c2a20757a

**Comment by ItsDrike** [comment]
Fixed in c38652c

**Comment by structnull** [comment]
Dude that was fast 🫡 can't thank you enough.

closing the issue as the plugin is working perfectly as intended
